src/gui/transaction.go

```diff
@@ -130,6 +138,97 @@ func getTransactionByID(gate *daemon.Gateway) http.HandlerFunc {
 	}
 }
 
+// Returns transactions that match the filters.
+// Method: GET
+// URI: /transactions
+// Args:
+//     addrs: Comma seperated addresses [optional, returns all transactions if no address provided]
+//     confirmed: Whether the transactions should be confirmed [optional, must be 0 or 1; if not provided, returns all]
+func getTransactions(gateway *daemon.Gateway) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		if r.Method != http.MethodGet {
+			wh.Error405(w)
+			return
+		}
+
+		// Gets 'addrs' parameter value
+		addrs, err := parseAddressesFromStr(r.FormValue("addrs"))
+		if err != nil {
+			wh.Error400(w, fmt.Sprintf("parse parament: 'addrs' failed: %v", err))
+			return
+		}
+
+		// Initialize transaction filters
+		flts := []visor.TxFilter{visor.AddrsFilter(addrs)}
+
+		// Gets the "confirmed" parameter value
+		confirmedStr := r.FormValue("confirmed")
+		if confirmedStr != "" {
+			confirmed, err := parseConfirmedFromStr(confirmedStr)
+			if err != nil {
+				wh.Error400(w, fmt.Sprintf("parse paramenter: 'confirmed' failed: %v", err))
+				return
+			}
+
+			flts = append(flts, visor.ConfirmedTxFilter(confirmed))
+		}
+
+		// Gets transactions
+		txns, err := gateway.GetTransactions(flts...)
+		if err != nil {
+			logger.Error("get transactions failed: %v", err)
+			wh.Error500(w)
+			return
+		}
+
+		// Converts visor.Transaction to visor.TransactionResult
+		txRlts, err := visor.NewTransactionResults(txns)
+		if err != nil {
+			logger.Error("Converts []visor.Transaction to visor.TransactionResults failed: %v", err)
+			wh.Error500(w)
+			return
+		}
+
+		wh.SendOr404(w, txRlts.Txns)
+	}
+}
+
+// Parses comma seperated addresses string into []cipher.Address,
+func parseAddressesFromStr(addrStr string) ([]cipher.Address, error) {
+	if addrStr == "" {
+		return nil, nil
+	}
+
+	var addrs []cipher.Address
+	for _, as := range strings.Split(addrStr, ",") {
+		s := strings.TrimSpace(as)
+		if s == "" {
+			continue
+		}
+
+		a, err := cipher.DecodeBase58Address(s)
+		if err != nil {
+			return nil, err
+		}
+
+		addrs = append(addrs, a)
+	}
+
+	return addrs, nil
+}
+
+// Parses the confirmed string int bool value, must be "0" or "1".
```

Comment:
Make a `parseBool` method

`"0", "f", "F", "false", "FALSE", ""` are false
`"1", "t", "T", "true", "TRUE"` are true

will use this for other APIs

Reply:
ok

Reply:
nevermind, can rewrite this to use `strconv.ParseBool`

Reply:
https://golang.org/pkg/strconv/#ParseBool